src/client.ts

```diff
@@ -282,7 +289,7 @@ export default class Client {
     const useYjit = vscode.workspace.getConfiguration("rubyLsp").get("yjit");
 
     // Enabling YJIT only provides a performance benefit on Ruby 3.2.0 and above
-    if (!useYjit || env.RUBY_VERSION! < "3.2.0") {
+    if (!useYjit || !this.ruby.rubyVersion || this.ruby.rubyVersion < "3.2.0") {
```

Comment:
I think we should use proper YJIT feature detection and not do Ruby version comparisons here. Something like `this.ruby.yjit_enabled` would be much nicer. It is as simple as checking if [`RubyVM::YJIT`](https://ruby-doc.org/core-3.1.2/RubyVM/YJIT.html) is defined using `defined?(RubyVM::YJIT)`

Reply:
Didn't know about the `RubyVM::YJIT` constant, thanks for the tip. For the version checking, I tried avoiding a new dependency and just compared the major and minor versions. Let me know what you think!